feat(winbar): add configurable control buttons (#44)
igorlfs/nvim-dap-view: README.md

![console](https://github.com/user-attachments/assets/0980962c-e3da-4f16-af4c-786ef7fa4b18)

> You can also enable the controlbar which exposes some clickable buttons by settings `winbar.controls.enable`.

Reviewer: ```suggestion
You can also enable the control bar which exposes some clickable buttons by settings `winbar.controls.enable`.
```

The position can be configured with `winbar.controls.position` and the enabled buttons with `winbar.controls.buttons`. 

> The icons can be configured with `winbar.controls.icons`. 

Reviewer: Instead of listing (almost) all options available, we could be more brief. Something along the lines of "The control bar is fully customizable, checkout the options on default configuration section".

NvimDapViewSeparator
NvimDapViewThread
NvimDapViewThreadStopped
NvimDapViewTab
NvimDapViewTabSelected
NvimDapViewControlNC
NvimDapViewControlPlay
NvimDapViewControlPause
NvimDapViewControlStepInto
NvimDapViewControlStepOut
NvimDapViewControlStepOver
NvimDapViewControlStepBack
NvimDapViewControlRunLast
NvimDapViewControlTerminate

> NvimDapViewControlDisconnect

Reviewer: This code block is getting big! Do you mind wrapping it in a `<details>` tag? It also makes sense to add the default links (and use a table instead), but I can do that later.

Author: Done

</details>

> ### Custom buttons

Reviewer: Can you update the ToC to also include this topic? I'm frequently forgetting about it myself, I wonder if there's some way to automate it

Author: Updated 👍🏼 

I think there might be some action to do that.
Maybe at some point, when the documentation grows, these kinds of small tasks become an argument for a dedicated documentation site (and IMHO, it could be a builtin GitHub feature)

Reviewer: >  an argument for a dedicated documentation site

loved the idea!

> `nvim-dap-view` provides some default buttons, but you can also add your own. To do that, in the `controls` table you can use the `custom_buttons` table to declare your new button and then add it at the position you want in the `buttons` list. 

Reviewer: ```suggestion
`nvim-dap-view` provides some default buttons for the control bar, but you can also add your own. To do that, in the `controls` table you can use the `custom_buttons` table to declare your new button and then add it at the position you want in the `buttons` list. 
```

A custom button has 2 methods:

- `render` returning a string used to display the button (typically an emoji or a NerdFont glyph wrapped in an highlight group)
- `action` a function that will be executed when the button is clicked. The function receives 3 arguments:
    - `clicks` the number of clicks
    - `button` the button clicked (`l`, `r`, `m`)
    - `modifiers` a string with the modifiers pressed (`c` for `control`, `s` for `shift`, `a` for `alt` and `m` for `meta`)

See the `@ N` section in `:help statusline` for the complete specifications of a click handler.

<details>
    <summary>Example custom buttons</summary>

An example adding 2 buttons:
- `fun`: the most basic button possible, just prints "🎊" when clicked

> - `term_restart`: an hybrid button that acts as a stop/restart button. If the stop button is triggered by anything else than a click (middle click, right click, double click or click with a modifier), it will disconnect the session instead.

Reviewer: ```suggestion
- `term_restart`: an hybrid button that acts as a stop/restart button. If the stop button is triggered by anything else than a single left click (middle click, right click, double click or click with a modifier), it will disconnect the session instead.
```

```lua
winbar = {
  controls = {
    enabled = true,
    buttons = {"play", "step_into", "step_over", "step_out", "term_restart", "fun" },
    custom_buttons = {
      fun = {
        render = function() return "🎉" end,
        action = function() vim.print("🎊") end,
      },
      -- Stop/Restart button

>       -- Double clic, middle clic or clic with a modifier disconnect instead of stop

Reviewer: ```suggestion
      -- Double click, middle click or click with a modifier disconnect instead of stop
```
